web/docs/understand/operators/transformations/shell.md

```diff
@@ -0,0 +1,68 @@
+# shell
+
+Executes a system command and hooks its raw stdin and stdout into the pipeline.
+
+## Synopsis
+
+```
+shell <command>
+```
+
+## Description
+
+The `shell` operator forks the process and executes the provided command.
+Thereafter, it connects the child's stdin to the operator's input, and the
+child's stdout to the operator's output. When `shell` receive new bytes as
+input, it writes them to the child's standard input. In parallel, `shell`
+attempts to read from the child's stdout and copies new bytes into the operator
+output.
+
+### `<command>`
+
+The command to execute and hook into the pipeline processing.
+
+The value of `command` is a single string. If you would like to pass a command
+line as you would on the shell, use single or double quotes for escaping, e.g.,
+`shell 'jq -C'` or `shell "jq -C"`.
+
+## Examples
+
+Consider the use case of converting CSV to JSON:
+
+```bash
+vast exec 'read csv | write json' | jq
+```
+
+The `write json` operator produces NDJSON. Piping this output to `jq` generates a
+colored, tree-structured variation that is (arguably) easier to read. Using the
+`shell` operator, you can integrate Unix tools that rely on
+stdin/stdout for input/output as "native" operators that process raw bytes. For
+example, in this pipeline:
+
+```
+print json | save stdout
+```
+
+The [`print`](../transformations/print.md) operator produces raw bytes and
+[`save`](../sinks/save.md) accepts raw bytes. The `shell` operator therefore
+fits right in the middle:
+
+```
+print json | shell jq | save stdout
```

Comment:
This example feels a bit useless, can you use something else here? I liked the Zeek example a lot that you posted in Discord earlier.

Reply:
I added a more elaborate example. We could consider ditching the `jq` as well, but I do want to start with a simple one to explain the mechanics.

Reply:
@dominiklohmann just to confirm, the function I need to implement is this one?

```cpp
  auto operator()(generator<std::monostate> input,
                  operator_control_plane& ctrl) const -> generator<chunk_ptr> {
    // TODO: implement
  }
```

I wonder how I can reuse some of the coroutine code from the other function. Maybe with a templated helper function with the same signature, but one that does `if constexpr` on the first argument? Then I could dispatch on `std::monostate` vs `chunk_ptr`. It might get convoluted, but I could do the same on the return value for a `shell` sink.